src/components/App.js

```diff
@@ -40,63 +50,149 @@ function App(props) {
     setRegisterPopupOpen(true);
   }
 
+  function handleTokenCheck() {
+    const jwt = localStorage.getItem("jwt");
+    if (jwt) {
+      // we're checking the user's token
+      mainApi
+        .checkToken(jwt)
+        .then((res) => {
+          if (res) {
+            loadUserAndArticlesList();
+            setIsLoggedIn(true);
+            props.history.push("/");
+          }
+        })
+        .catch((err) => {
+          console.log(`Error: ${err}`);
+        });
+    }
+  }
+
+  function loadUserAndArticlesList() {
+    Promise.all([mainApi.getUserInformation(), mainApi.getArticles()])
+      .then(([data, articles]) => {
+        //load user info
+        setCurrentUser(data.data);
+        //load initial cards list
+        setSavedNewsList(articles.data);
+      })
+      .catch((err) => {
+        console.log(`Error: ${err}`);
+      });
+  }
+
   function handleLogin({ email, password }) {
-    const user = userList.find(
-      (element) => element.email === email && element.password === password
-    );
-    if (user) {
-      setCurrentUser({
-        name: user.name,
-        email: user.email,
+    mainApi
+      .login(email, password)
+      .then((res) => {
+        if (res) {
+          localStorage.setItem("jwt", res.token);
+          handleTokenCheck();
+          closeAllPopups();
+        }
+      })
+      .catch((err) => {
+        closeAllPopups();
+        setIsPopupMessageOpen(true);
+        setIsPopupMessage("Invalid Login");
       });
-      setIsLoggedIn(true);
-      closeAllPopups();
-      props.history.push("/");
-    } else {
-      closeAllPopups();
-      setIsPopupMessageOpen(true);
-      setIsPopupMessage("Invalid Login");
-    }
   }
 
   function handleLogout() {
-    setCurrentUser({});
     setIsLoggedIn(false);
+    setCurrentUser({});
+    localStorage.removeItem("jwt");
     props.history.push("/");
   }
 
   function handleRegister({ email, username, password }) {
-    const user = userList.find((element) => element.email === email);
-    if (user) {
-      setIsPopupMessageOpen(true);
-      setIsPopupMessage("Invalid Registration");
-    } else {
-      userList.push({ email: email, name: username, password: password });
-      closeAllPopups();
-      setIsPopupMessageOpen(true);
-      setIsPopupMessage("Registration successfully completed!");
+    mainApi
+      .register(email, password, username)
+      .then((res) => {
+        if (res) {
+          closeAllPopups();
+          setIsPopupMessageOpen(true);
+          setIsPopupMessage("Registration successfully completed!");
+        }
+      })
+      .catch((err) => {
+        setIsPopupMessageOpen(true);
```

Comment:
`          setIsPopupMessageOpen(true);
`
**Could be improved**



it's better to add it only in `finally` not to duplicate it in `then` and `catch`